### ISSUE TITLE
NTLM Proxy Auth for the HTTP client

### ISSUE THREAD

**Issue by stringy05** [dropwizard/dropwizard#2084]
I'm doing some work at BigCorp who use Windows and NTLM proxies everywhere,  so I need my client to be able to authenticate against a proxy using NTLM.

I've written a patch for it in dropwizard-client, basic outline is:
- Change the AuthConfiguration, adding NTLM requrired fields such as 
  - realm
  - domain
  - hostname
  - a credentialType parameter, which determines which implementation of the Credential interface to use

The alter the proxy creation part of the createClient method to provide a NTLM aware 

``` 
    if (auth != null) {
        if (credentialsProvider == null) {
            credentialsProvider = new BasicCredentialsProvider();
        }
        // set the AuthScope
        AuthScope authScope = new AuthScope(httpHost, auth.getRealm(), auth.getAuthScheme());

        // set the credentials type
        Credentials credentials = null;
        if (null != auth.getCredentialType() && auth.getCredentialType().equalsIgnoreCase(AuthConfiguration.NT_AUTH_TYPE)) {
            credentials = new NTCredentials(auth.getUsername(), auth.getPassword(), auth.getHostname(), auth.getDomain());
        } else {
            credentials = new UsernamePasswordCredentials(auth.getUsername(), auth.getPassword());
        }

        credentialsProvider.setCredentials(authScope, credentials);
    }
} 
```

Works on my machine :)

**Comment by arteam** [comment]
This looks like a useful addition! I would accept the patch.

**Comment by arteam** [comment]
Closed via #2091
